pkg/platform/interface.go

```diff
@@ -2,14 +2,26 @@ package platform
 
 import (
 	"context"
-
-	"go.keploy.io/server/pkg/models"
 )
 
 type TestCaseDB interface {
-	WriteTestcase(tc *models.TestCase, ctx context.Context, filters *models.Filters) error
-	WriteMock(tc *models.Mock, ctx context.Context) error
+	WriteTestcase(tc MockDescriptor, ctx context.Context, filters MockDescriptor) error
+	WriteMock(tc MockDescriptor, ctx context.Context) error
+
+	ReadTestcase(path string, lastSeenId MockDescriptor, options MockDescriptor) ([]MockDescriptor, error)
+	ReadTcsMocks(tc MockDescriptor, path string) ([]MockDescriptor, error)
+	ReadConfigMocks(path string) ([]MockDescriptor, error)
+}
+
+type TestReportDB interface {
+	Lock()
+	Unlock()
+	SetResult(runId string, test MockDescriptor)
+	GetResults(runId string) ([]MockDescriptor, error)
+	Read(ctx context.Context, path, name string) (MockDescriptor, error)
+	Write(ctx context.Context, path string, doc MockDescriptor) error
+}
 
-	ReadTestcase(path string, options interface{}) ([]*models.TestCase, error)
-	ReadMocks(path string) ([]*models.Mock, []*models.Mock, error)
+type MockDescriptor interface {
```

Comment:
Since this interface is also used for test-reports and testcase, maybe it can be a little generic like: "dataDesriptor", "KindSpecifier".